src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -517,6 +517,43 @@ private static void ArrayTypeCheck_Helper(object obj, void* elementType)
             }
         }
 
+        // Helpers for boxing
+        [DebuggerHidden]
+        internal static object? Box_Nullable(MethodTable* srcMT, ref byte nullableData)
+        {
+            Debug.Assert(srcMT->IsNullable);
+
+            // If 'hasValue' is false, return null.
+            if (!Unsafe.As<byte, bool>(ref nullableData))
```

Comment:
```suggestion
            if (nullableData == 0)
```

Reply:
Ok, you can ignore this suggestion. (I was comparing this with NAOT impl that uses the other style.)

Reply:
It was used in Rebox which is now refactored in this PR, so I have aligned it with NativeAOT.